### PR TITLE
Add rule S2068 No hardcoded credentials

### LINKED ISSUE #25: Rule S2068: Credentials should not be hardcoded
Implements [RSPEC-2068](https://jira.sonarsource.com/browse/RSPEC-2068)

### PR DESCRIPTION
**Pull request by saberduck**
Fix #25

### CODE REVIEW

File: common-rule-engine/src/main/java/org/sonar/commonruleengine/checks/NoHardcodedCredentialsCheck.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+  public NoHardcodedCredentialsCheck() {
+    super(UastNode.Kind.ASSIGNMENT);
+  }
+
+  @Override
+  public void initialize(EngineContext context) {
+    super.initialize(context);
+    String[] words = credentialWords.toLowerCase().split(",");
+    String pattern = Arrays.stream(words).collect(Collectors.joining("|"));
```
Comment (ghost) [inline common-rule-engine/src/main/java/org/sonar/commonruleengine/checks/NoHardcodedCredentialsCheck.java]: Shouldn't we call `Pattern.quote` on each word?

File: common-rule-engine/src/main/java/org/sonar/uast/UastNode.java [hunk 2]
```diff
@@ -90,6 +91,21 @@ public UastNode firstToken() {
     return null;
   }
 
+  public String tokenize() {
```
Comment (ghost) [inline common-rule-engine/src/main/java/org/sonar/uast/UastNode.java]: I was surprised by the name of this method and what it does. (I expect tokenization to _split_ something to tokens, not to _join_ tokens to a single value.) I don't have a better name in mind :)

File: common-rule-engine/src/test/resources/org/sonar/commonruleengine/checks/go/NoHardcodedCredentialsCheck/NoHardcodedCredentialsCheck.go [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+func test()  {
+	pwd := "supersecret" // Noncompliant
+	myPassword := "supersecret" // Noncompliant
+	url := "login=user&passwd=secret" // Noncompliant
+}
+
+func explicitDecl()  {
+	var pwd string
+	pwd = "secret"
```
Comment (ghost) [inline common-rule-engine/src/test/resources/org/sonar/commonruleengine/checks/go/NoHardcodedCredentialsCheck/NoHardcodedCredentialsCheck.go]: Why is this compliant?

**Review by ghost — APPROVED** [review]

File: common-rule-engine/src/main/java/org/sonar/uast/UastNode.java (hunk 2 above)
Comment (saberduck) [reply common-rule-engine/src/main/java/org/sonar/uast/UastNode.java]: Indeed, I share the feeling. I will rename method to `joinTokens` , it's better to be explicit

File: common-rule-engine/src/test/resources/org/sonar/commonruleengine/checks/go/NoHardcodedCredentialsCheck/NoHardcodedCredentialsCheck.go (hunk 3 above)
Comment (saberduck) [reply common-rule-engine/src/test/resources/org/sonar/commonruleengine/checks/go/NoHardcodedCredentialsCheck/NoHardcodedCredentialsCheck.go]: This is a mistake, thanks! uast file was not in sync with Go source.

**Pull request merged**